### PR TITLE
Burnt-out flashes are more obvious

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request
Flashes update in-hand sprite, name and description to reflect being burnt out.

_Not the most readable sprite, but it's on the left_

## Why It's Good For The Game

Makes it far easier to find useful flashes in piles (the broken overlay doesn't show in the context menu) and recognise whether someone is holding a functional flash or not. 

Fixes a longstanding issue where in-hand sprites for flashes look functional even after they've burnt out.

## Changelog
:cl: cacogen
qol: The flash now has in-hand sprites for when the bulb is blown and the name and description update to reflect it too
/:cl:

### CODE REVIEW

File: code/modules/assembly/flash.dm [hunk 1]
```diff
@@ -71,13 +80,13 @@
 /obj/item/assembly/flash/proc/burn_out() //Made so you can override it if you want to have an invincible flash from R&D or something.
 	if(!burnt_out)
 		burnt_out = TRUE
+		if(ismob(loc))
+			var/mob/M = loc
+			M.visible_message("<span class='danger'>[src] burns out!</span>","<span class='userdanger'>[src] burns out!</span>")
+		else
+			var/turf/T = get_turf(src)
+			T.visible_message("<span class='danger'>[src] burns out!</span>")
```
Comment (Mothblocks) [inline code/modules/assembly/flash.dm:88]: I know you didn't write this, but `visible_message` is something on `/atom`, this should just be `loc?.visible_message`

**Comment by cacogen** [comment]
@mothblocks Is this what you meant

File: code/modules/assembly/flash.dm [hunk 2]
```diff
@@ -80,12 +80,7 @@
 /obj/item/assembly/flash/proc/burn_out() //Made so you can override it if you want to have an invincible flash from R&D or something.
 	if(!burnt_out)
 		burnt_out = TRUE
-		if(ismob(loc))
-			var/mob/M = loc
-			M.visible_message("<span class='danger'>[src] burns out!</span>","<span class='userdanger'>[src] burns out!</span>")
-		else
-			var/turf/T = get_turf(src)
-			T.visible_message("<span class='danger'>[src] burns out!</span>")
+		loc?.visible_message("<span class='danger'>[src] burns out!</span>","<span class='userdanger'>[src] burns out!</span>")
```
Comment (cacogen) [inline code/modules/assembly/flash.dm:83]: @mothblocks Is this what you meant

Comment (Mothblocks) [reply code/modules/assembly/flash.dm:83]: yeah

**Review by Twaticus — APPROVED** [review]

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**